File: 백준/Bronze/10769. 행복한지 슬픈지/행복한지 슬픈지.py
import sys
input = sys.stdin.readline

string = input().rstrip()

happy_cnt = string.count(':-)')
sad_cnt = string.count(':-(')

if happy_cnt == 0 and sad_cnt == 0: print('none')
elif happy_cnt == sad_cnt: print('unsure')
elif happy_cnt > sad_cnt: print('happy')
else: print('sad')

File: 백준/Bronze/10769. 행복한지 슬픈지/행복한지 슬픈지.c
#include <stdio.h>
#include <string.h>

char str[260];
int main(void) {
    gets(str);
    int happy = 0;
    int sad = 0;

    for (int i = 0; i < strlen(str) - 2; i++) {
        if (str[i] == ':' && str[i+1] == '-' && str[i+2] == ')') happy += 1;
        if (str[i] == ':' && str[i+1] == '-' && str[i+2] == '(') sad += 1;
    }

    if (happy == 0 && sad == 0) printf("none");
    else if (happy == sad) printf("unsure");
    else if (happy > sad) printf("happy");
    else printf("sad");
    return 0;
}